src/ray/core_worker/common.h

```diff
@@ -49,19 +50,19 @@ class TaskArg {
   }
 
   /// Get the value.
-  std::shared_ptr<Buffer> GetValue() const {
-    RAY_CHECK(data_ != nullptr) << "This argument isn't passed by value.";
-    return data_;
+  const RayObject &GetValue() const {
+    RAY_CHECK(value_ != nullptr) << "This argument isn't passed by value.";
```

Comment:
Let's try not to use `RAY_CHECK` in the core worker as it causes an `Abort trap: 6`, making it very inconvenient to debug (and horrible for users if it happens somehow). Better to use `RAY_RETURN_NO_OK`.

Reply:
How is that possible? `RAY_RETURN_NOT_OK` is used for functions with the return type of `Status`.

Reply:
This `RAY_CHECK` should not get triggered by user code. If it fails, it must be a bug in ray codebase.

Reply:
Yeah, I think it's better to use `RAY_CHECK` to catch system bugs.

Reply:
Oops, sorry - of course you can't use `RAY_RETURN_NOT_OK`. Actually, isn't it better to define separate classes for by value and by reference arguments instead of having these checks? That way the compiler can catch these bugs for us...